File: go/src/leetcode/editor/cn/0350_intersection_of_two_arrays_ii.go
package main

import "sort"

func main() {
	println("350、intersection_of_two_arrays_ii")
}

// leetcode submit region begin(Prohibit modification and deletion)
func intersect(nums1 []int, nums2 []int) []int {
	sort.Ints(nums1)
	sort.Ints(nums2)
	left, right, res := 0, 0, []int{}
	for left < len(nums1) && right < len(nums2) {
		x, y := nums1[left], nums2[right]
		if x == y {
			res = append(res, x)
			left++
			right++
		} else if x < y {
			left++
		} else {
			right++
		}
	}
	return res
}

//leetcode submit region end(Prohibit modification and deletion)

File: rust/src/leetcode/editor/cn/_0350_intersection_of_two_arrays_ii.rs
#[cfg(test)]
mod tests {
    #[allow(unused_imports)]
    use super::*;

    #[test]
    fn test() {
        println!("350、intersection_of_two_arrays_ii");
    }
}

#[allow(dead_code)]
struct Solution;

//leetcode submit region begin(Prohibit modification and deletion)
impl Solution {
    #[allow(dead_code)]
    pub fn intersect(mut nums1: Vec<i32>, mut nums2: Vec<i32>) -> Vec<i32> {
        nums1.sort_unstable();
        nums2.sort_unstable();
        let (mut left, mut right, mut res) = (0, 0, vec![]);
        while left < nums1.len() && right < nums2.len() {
            let (x, y) = (nums1[left], nums2[right]);
            if x == y {
                res.push(x);
                left += 1;
                right += 1;
            } else if x < y {
                left += 1;
            } else {
                right += 1;
            }
        }
        return res;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
